web/sendgrid_webhooks.py

```
@@ -0,0 +1,188 @@
+            return HttpResponseForbidden()
+
+        if not event_verifier.verify_signature(request.body, signature, timestamp, public_key):
+            logger.warning("SendGrid webhook signature verification failed")
+            return HttpResponseForbidden()
+    except Exception:
+        logger.exception("Error verifying SendGrid webhook signature")
+        return HttpResponseForbidden()
+    try:
+        # Parse the incoming JSON
+        data = json.loads(request.body)
+        logger.info(f"Received SendGrid webhook with {len(data)} events")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)